nuget/lib/dependabot/nuget/file_updater/project_file_declaration_finder.rb

```diff
@@ -95,6 +111,80 @@ def declaring_file
 
           raise "No file found with name #{filename}!"
         end
+
+        def sdk_import_strings
+          sdk_strings(SDK_IMPORT_REGEX, "Import", "Sdk", "Version")
+        end
+
+        # rubocop:disable Metrics/PerceivedComplexity
+        def sdk_project_strings
+          dep_name = dependency_name&.downcase
+          dep_version = declaring_requirement.fetch(:requirement)
+          strings = []
+          declaring_file.content.scan(SDK_PROJECT_REGEX).each do |string|
+            xml = string
+            xml += "</Project>" unless string.end_with?("/>")
+            node = Nokogiri::XML(xml)
+            node.remove_namespaces!
+            element = node.at_xpath("/Project")
+            next unless element
+
+            sdk_references = element.attribute("Sdk")&.value ||
+                             element.attribute("sdk")&.value
+            sdk_references = sdk_references&.strip
+            next unless sdk_references&.include?("/")
+
+            sdk_references.split(";").each do |sdk_reference|
+              parts = sdk_reference.split("/")
+              next unless parts.length == 2
+              next unless parts[0]&.downcase == dep_name
+              next unless parts[1] == dep_version
+
+              strings << sdk_reference
+            end
+          end
+          # Drop duplicates, to avoid potential issues with repeated replacements (e.g. upgrading Foo/1.0.0-beta to
+          # Foo/1.0.0-beta.2). However, that turns the empty array into nil.
+          strings.uniq! || []
```

Comment:
```suggestion
          strings.uniq
```

The `uniq!` version of the method mutates the array in-place, but since we're returning the value it's fine to return a new instance, and this so called "non bang" version of the method does not return an empty array into nil.

Reply:
Applied